cmd/snap-update-ns/entry.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces/mount"
+	"github.com/snapcore/snapd/osutil"
+)
+
+// XSnapdMode returns the file mode associated with x-snapd.mode mount option.
+// If the mode is not specified explicitly then a default mode of 0755 is assumed.
+func XSnapdMode(e *mount.Entry) (os.FileMode, error) {
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd.mode=") {
+			kv := strings.SplitN(opt, "=", 2)
+			var mode os.FileMode
+			n, err := fmt.Sscanf(kv[1], "%o", &mode)
+			if err != nil || n != 1 {
+				return 0, fmt.Errorf("cannot parse octal file mode from %q", kv[1])
+			}
+			return mode, nil
+		}
+	}
+	return 0755, nil
+}
+
+// XSnapdUid returns the user associated with x-snapd-user mount option.
+// If the mode is not specified explicitly then a default "0" (root) user is
+// returned.
+func XSnapdUid(e *mount.Entry) (uid uint64, err error) {
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd.uid=") {
+			kv := strings.SplitN(opt, "=", 2)
+			uid, err = osutil.FindUid(kv[1])
```

Comment:
This and XsnapGid do not support specifying by uid number (ie, `x-snapd.uid=123` will fail). If this is intentional, can you please add a comment?

Reply:
Interesting. Since numbers are not valid user or group names we could extend this code to just return that number directly. I don't think we need this just yet but it's a natural extension to how this could work.

Reply:
On second thought I implemented this now in c3ec78e0d79bd583bf5892377c2c85383ff0635e